Lesson18/gokce/script.js

```diff
@@ -0,0 +1,247 @@
+- Logs each recipe from recipes in a nice format:
+  Name: Pasta
+  Ingredients: pasta, tomato, garlic
+  Cooking Time: 20 minutes
+*/
+
+function displayAllRecipes() {
+  for (let recipe of recipes) {
+    console.log(`Name:${recipe.name}`);
+    console.log(`Ingredients:${recipe.ingredients.join(",")}`);
+    console.log(`Cooking Time:${recipe.cookingTimeMinutes} minutes`);
+  }
+}
+
+displayAllRecipes(recipes);
+
+/*
+-----------------------------------------------------------
+  STEP 3: Add a New Recipe
+-----------------------------------------------------------
+Function: addRecipe(name, ingredients, cookingTime)
+- Checks if a recipe with the same name exists.
+- If yes, log a warning and return.
+- If not, add the new recipe and log success.
+- ingredients should be an array like ['egg', 'milk', 'flour']
+*/
+function addRecipe(name, ingredients, cookingTime) {
+  if (
+    typeof name !== "string" ||
+    !Array.isArray(ingredients) ||
+    typeof cookingTime !== "number"
+  ) {
+    console.log(
+      `Please provide a string format for "name", an array for "ingredients", and a number for "cookingTime".`,
+    );
+    return;
+  }
+  for (let recipe of recipes) {
+    if (name === recipe.name) {
+      console.warn(`This recipe already exists.`);
+      return;
+    }
+  }
+  const newRecipe = {
+    name: name,
+    ingredients: ingredients,
+    cookingTimeMinutes: cookingTime,
+  };
+  recipes.push(newRecipe);
+  console.log(`Success`);
+}
+
+addRecipe("Milk Pudding", ["milk", "sugar", "starch", "vanilla"], 15);
+addRecipe("Banana Smoothie", ["banana", "milk", "honey"], 10);
+/*
+-----------------------------------------------------------
+  STEP 4: View a Recipe by Name
+-----------------------------------------------------------
+Function: viewRecipe(name)
+- Looks for the recipe by name and logs all its info.
+- If not found, shows a message.
+*/
+function viewRecipe(name) {
+  if (typeof name !== "string") {
+    console.log(`Please provide a string format for "name".`);
+    return;
+  }
+  if (name !== recipes.name) {
+    console.log(`Recipe not found`);
+  }
+  for (let recipe of recipes) {
+    if (name === recipe.name) {
+      console.log(`You can view the recipe for ${name}:
+      Ingredients: ${recipe.ingredients.join(",")}
+      Cooking Time: ${recipe.cookingTimeMinutes} minutes`);
+    }
+  }
+}
+viewRecipe("3-Ingredient Quick Bread");
+viewRecipe("Chocolate Cake");
+/*
+-----------------------------------------------------------
+  STEP 5: Update a Recipe
+-----------------------------------------------------------
+Function: updateRecipe(name, newIngredients, newCookingTime)
+- Finds the recipe by name.
+- Updates ingredients and cookingTime.
+- Logs success or error message.
+*/
+
+function updateRecipe(name, newIngredients, newCookingTime) {
+  if (
+    typeof name !== "string" ||
+    !Array.isArray(newIngredients) ||
+    typeof newCookingTime !== "number"
+  ) {
+    console.log(
```

Comment:
Since the message printed here is related to an actual error, you can use an error level log message instead: `console.error(`Please provide a string format for "name", an array for "newingredients", and a number for "newcookingTime".`)`